pkg/comp-functions/functions/vshnredis/release.go

```diff
@@ -194,18 +195,32 @@ func getReleaseValues(r *v1beta1.Release) (map[string]interface{}, error) {
 }
 
 // setReleaseVersion sets the version from the claim if it's a new instance otherwise it is managed by maintenance function
-func setReleaseVersion(comp *vshnv1.VSHNRedis, values map[string]interface{}, observed map[string]interface{}) error {
-	var err error
+func setReleaseVersion(ctx context.Context, comp *vshnv1.VSHNRedis, values map[string]interface{}, observed map[string]interface{}) error {
+	l := controllerruntime.LoggerFrom(ctx)
+
 	tag, _, err := unstructured.NestedString(observed, "image", "tag")
 	if err != nil {
 		return fmt.Errorf("cannot get image tag from values in release: %v", err)
 	}
-	if tag != "" {
-		// In case the tag is set, keep the observed version
-		err = unstructured.SetNestedField(values, tag, "image", "tag")
-	} else {
-		// In case the tag is not set then this is a new Release therefore we need to set the version from the claim
-		err = unstructured.SetNestedField(values, comp.Spec.Parameters.Service.Version, "image", "tag")
+
+	desiredVersion, err := semver.ParseTolerant(comp.Spec.Parameters.Service.Version)
+	if err != nil {
+		l.Info("failed to parse desired redis version", "version", comp.Spec.Parameters.Service.Version)
+		// If the desired version is not parsable, just keep the observed one
```

Comment:
This scenario should probably be reported to the user somehow (eg with an event or something)

Reply:
Well this should never happen because they can only choose between `6.2` and `7.0`, but you're right, it's probably better to just return an error and then they issue should bubble up to the user

Reply:
LGTM now